rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -2148,8 +2155,136 @@ def make_new_set():
+                id_type = header
+            else:
+                _logger.error('File header must be sample_id or biobank_id')
+                return 1
+
+            for row in csv_reader:
+                id_list.append(row[0])
+
+            if id_type == 'sample_id':
+                set_members = member_dao.get_members_from_sample_ids(id_list)  # will not return results in IGNORE state
+            else:
+                set_members = member_dao.get_members_from_biobank_ids(id_list)
+            for set_member in set_members:
+                if set_member.ignoreFlag == 1:
+                    continue
+                if unblock_results:
+                    set_member.blockResults = 0
+                if unblock_research:
+                    set_member.blockResearch = 0
+                if not self.args.dryrun:
+                    member_dao.update(set_member)
+                else:
+                    _logger.info(f"Will unblock genomic_set_member {set_member.id}")
+                if id_type == 'biobank_id' and set_member.sampleId is None:
+                    # attempt aw1 ingestion
+                    try:
+                        pre = server_config[config.BIOBANK_ID_PREFIX][0]
+                    except KeyError:
+                        # Set default for unit tests
+                        pre = "A"
+                    bid = f"{pre}{set_member.biobankId}"
+                    result = aw1_raw_dao.get_raw_record_from_identifier_genome_type(identifier=bid,
+                                                                                    genome_type=set_member.genomeType)
+                    if result is not None:
+                        if not self.args.dryrun:
+                            self._ingest_member('AW1_MANIFEST', server_config, set_member)
+                        else:
+                            _logger.info(f"Will try AW1 ingestion for {set_member.id}")
+                    else:
+                        no_aw1_data.append(set_member.biobankId)
+                vmetric = metrics_dao.get_metrics_by_member_id(set_member.id)
+                if vmetric is None:
+                    # Look up AW2 data
+                    aw2_record = aw2_raw_dao.get_raw_record_from_identifier_genome_type(
+                        identifier=set_member.sampleId, genome_type=set_member.genomeType)
+                    if aw2_record is not None:
+                        # Perform AW2 ingestion
+                        if not self.args.dryrun:
+                            self._ingest_member('METRICS_INGESTION', server_config, set_member)
+                        else:
+                            _logger.info(f"Will try AW2 ingestion for {set_member.id}")
+                    else:
+                        no_aw2_data.append(set_member.sampleId)
+                else:
+                    if set_member.genomeType == 'aou_array':
+                        if vmetric.idatRedReceived != 1 or vmetric.idatGreenReceived != 1 \
```

Comment:
You can use `if any()` here instead of a series of `or` clauses. Also, check out `rdr_service/genomic/genomic_mappings.py:137-210` and `rdr_service/genomic/genomic_job_components.py:2460` you can use that pattern to check for required files.